packages/dx-grid-core/src/plugins/filtering-state/computeds.ts

```diff
@@ -0,0 +1,16 @@
+import { FilterExpression, Filter } from '../../types';
+import { PureComputed } from '@devexpress/dx-core';
+
+export const filterExpression: PureComputed<
+  [Filter[], FilterExpression], FilterExpression
+> = (filters, expression) => {
+  // tslint:disable-next-line: no-object-literal-type-assertion
+  const selfFilterExpr = { filters, operator: 'and' as 'and' } as FilterExpression;
```

Comment:
Shouldn't variable type be explicitly specified instead of type casting?
```js
const selfFilterExpr : FilterExpression = { filters, operator: 'and' };
```

Reply:
FilterExpression expects `filters` of Array type, so we need to cast it from ReadonlyArray, therefore `as` is used